en/file_formats/plan.md

```diff
@@ -1,93 +1,143 @@
-cruiseSpeed | 
-hoverSpeed | 
-items | The list of mission item objects associated with the mission.
-plannedHomePosition | The planned home position to show on the map when you are editing the mission. Values with array are latitude, longitude and altitude.
+--- | ---
+`version` | The version for the mission object. Current version is 2.
+`firmwareType` | The firmware type for which this mission was created. This is one of the [MAV_AUTOPILOT](https://mavlink.io/en/messages/common.html#MAV_AUTOPILOT) enum values. 
+`vehicleType` | The vehicle type for which this mission was created. This is one of the [MAV_TYPE](https://mavlink.io/en/messages/common.html#MAV_TYPE) enum values.
+`cruiseSpeed` | The default forward speed for Fixed wing or VTOL vehicles (i.e. when moving between waypoints).
+`hoverSpeed` | The default forward speed for multi-rotor vehicles.
+`items` | The list of mission item objects associated with the mission . The list may contain either/both [SimpleItem](#mission_simple_item) and [ComplexItem](#mission_complex_item) objects.
+`plannedHomePosition` | The planned home position is shown on the map and used for mission planning when no vehicle is connected. The array values shown above are (from top): latitude, longitude and AMSL altitude.
+
+The format of the simple and complex items is given below.
 
-## Mission Item - `SimpleItem`
+### `SimpleItem` - Simple Mission Item {#mission_simple_item}
 
-A simple item represents a single MAVLink [MISSION_ITEM](http://mavlink.org/messages/common#MISSION_ITEM) command.
+A simple item represents a single MAVLink [MISSION_ITEM](https://mavlink.io/en/messages/common.html#MISSION_ITEM) command.
 
 ```
-{
-    "autoContinue": true,
-    "command": 22,
-    "frame": 2,
-    "params": [
-        0,
-        0,
-        0,
-        0,
-        47.633127690000002,
-        -122.08867133,
-        50
-    ],
-    "type": "SimpleItem"
-},
+            {
+                "AMSLAltAboveTerrain": null,
+                "Altitude": 50,
+                "AltitudeMode": 0,
+                "autoContinue": true,
+                "command": 22,
+                "doJumpId": 1,
+                "frame": 3,
+                "params": [
+                    15,
+                    0,
+                    0,
+                    null,
+                    47.3985099,
+                    8.5451002,
+                    50
+                ],
+                "type": "SimpleItem"
+            }
 ```
 
-The values in a `SimpleItem` map directly to the values in [MISSION_ITEM](http://mavlink.org/messages/common#MISSION_ITEM):
+The field mapping is shown below.
 
 Key | Description
-----|------------
-autoContinue | MISSION_ITEM.autoContinue
-command | MISSION_ITEM.command
-frame | MISSION_ITEM.frame
-params | MISSION_ITEM.param1,2,3,4,x,y,z
+--- | ---
+`type` | `SimpleItem` for a simple item
+`AMSLAltAboveTerrain` | 
```

Comment:
What are the next three values? AMSLAltAboveTerrain is a bit unclear because an alt is ususally either AMSL or above terrain, Altitude similarly could be either, and you have a mode, but often the altitude might depend on the frame. Can you confirm how this works.

Reply:
A mission item has 7 params with it. Sometimes called 1-7, sometimes 1-4, then x/y/z since th last three are usually used for lat/lon/alt but not always.

Reply:
AMSLAltAboveTerrain is just this value shown to the user:
![screen shot 2018-08-02 at 11 16 49 am](https://user-images.githubusercontent.com/5876851/43602713-9a7a65e2-9645-11e8-9440-56fddd854b80.png)